keras_cv/layers/preprocessing/mosaic.py

```diff
@@ -164,8 +162,18 @@ def _batch_augment(self, inputs):
                     index,
                 ),
                 tf.range(batch_size),
+                fn_output_signature={
+                    "boxes": tf.RaggedTensorSpec(
+                        shape=[None, 4],
```

Comment:
This shape seems too restrictive -- what if we want to support higher dimension?

Reply:
Lets solve that if a need arises, but I think storing them in a different key makes sense for now?  I'd opt to follow the YAGNI principle here; it sounds like it could be useful, but lets wait until theres a strong need.